### PR TITLE
Add Swagger API Documentation

### LINKED ISSUE #12: Use Swagger to create API documentation
Integrate Swagger to document the backend API routes, including their request parameters, responses, and error codes. Keep the documentation updated as the API evolves.

**Reference:**
- [Swagger with Express](https://dev.to/franciscomendes10866/swagger-api-documentation-in-nodejs-express-2nkd)

### PR DESCRIPTION
**Pull request by swastik-apk**
## Description

This pull request introduces Swagger API documentation to the project and makes some minor improvements to the codebase. The most important changes include adding Swagger configuration and documentation for the API routes, updating dependencies, and ensuring consistent import styles.

### API Documentation:

* [`index.js`](diffhunk://#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346L1-R75): Added Swagger configuration and middleware to serve API documentation at `/api-docs`.
* [`routes/auth.js`](diffhunk://#diff-ac3a4f36715c2bdf5bdd58ed63b5a82c480840317e96c93b85e0888ee7bb5f31L2-L19): Added Swagger documentation for authentication routes including login, register, and change-password endpoints.
* [`routes/blog.js`](diffhunk://#diff-e8efed5855e6382e3615358e5079bb0e65fd6b7c3b800e331fc3259b044473b3L1-R162): Added Swagger documentation for blog routes including CRUD operations for blogs.
* [`routes/verify.js`](diffhunk://#diff-20fbac3109e5b3e655f65b364bc61ef97098707bc998e15217072563770a26b2L1-R42): Added Swagger documentation for the token verification endpoint.

### Dependency Updates:

* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L18-R25): Updated `express` to version 4.21.1 and added `swagger-jsdoc` and `swagger-ui-express` dependencies.

Closes #12 

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

#### Additional Information 
- Name: Swastik Bandyopadhyay
- Email: 24052040@kiit.ac.in
- Roll No: 24052040
- Branch: CSE

### CODE REVIEW

**Comment by swastik-apk** [comment]
@Tony-ArtZ @dead8309 Can you please review and accept the changes.

File: index.js [hunk 1]
```diff
@@ -1,56 +1,76 @@
...
 
-const PORT = process.env.PORT ?? 3000
+const PORT = process.env.PORT || 3000;
 
-// connect to the database
+// Connect to the database
 mongoose
-  .connect(process.env.MONGODB_URL, {
-    useNewUrlParser: true,
-    useUnifiedTopology: true,
```
Comment (dead8309) [inline index.js:44]: Don't delete the connection properties

**Comment by swastik-apk** [comment]
@dead8309 Updated the code. Check now.

**Comment by dead8309** [comment]
Thanks for your contribution!!

**Pull request merged**